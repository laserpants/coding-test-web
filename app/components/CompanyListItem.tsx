"use client";

import type { ICompany } from "../types/company";
import Image from "next/image";

export default function CompanyListItem({
  iconUrl,
  companyName,
  description,
}: ICompany) {
  return (
    <li className="flex items-center py-4">
      <Image
        src={iconUrl ?? "https://placehold.co/50/png"}
        alt={companyName}
        className="w-12 h-12 rounded-full"
        width="50"
        height="50"
      />
      <div className="ml-4 flex-1">
        <p className="text-base font-semibold text-gray-800">{companyName}</p>
        <div className="text-sm text-gray-500 line-clamp-2">
          <p className="text-sm text-gray-500 ">{description}</p>
        </div>
      </div>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth={2}
        stroke="currentColor"
        className="w-6 h-6 text-gray-400 ml-2 cursor-pointer"
      >
        <path strokeLinecap="round" strokeLinejoin="round" d="M9 5l7 7-7 7" />
      </svg>
    </li>
  );
}
