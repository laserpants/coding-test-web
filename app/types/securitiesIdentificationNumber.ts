/**
 * Represents an ISIN (International Securities Identification Number),
 * defined by ISO 6166. The format is a 12-character alphanumeric string:
 *
 * - 2 letters: country code (ISO 3166-1 alpha-2)
 * - 9 characters: security identifier
 * - 1 digit: checksum
 *
 * For more details, see: https://www.iso.org/standard/44866.html
 */
export type ISIN = string;

/**
 * Validates whether a given string is a valid ISIN (ISO 6166 format).
 *
 * @param isin - The ISIN string to validate.
 * @returns True if the ISIN is valid, or false otherwise.
 */
export function isValidISIN(isin: string): boolean {
  const isinRegex = /^[A-Z]{2}[A-Z0-9]{9}[0-9]$/;
  return isinRegex.test(isin);
}
