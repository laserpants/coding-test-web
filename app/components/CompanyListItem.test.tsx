import { render } from "@testing-library/react";
import { beforeEach, describe, it, expect, jest } from "@jest/globals";
import { act } from "react";
import CompanyListItem from "./CompanyListItem";
import { ICompany } from "../types/company";

global.fetch = jest.fn() as jest.MockedFunction<typeof fetch>;

describe("Company", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", async () => {
    const company = { companyName: "Company A" } as ICompany;

    const { container } = await act(async () =>
      render(<CompanyListItem {...company} />),
    );

    expect(container).toMatchSnapshot();
  });
});
