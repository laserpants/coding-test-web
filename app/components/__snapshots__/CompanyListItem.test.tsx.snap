// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company renders correctly 1`] = `
<div>
  <li
    class="flex items-center py-4"
  >
    <img
      alt="Company A"
      class="w-12 h-12 rounded-full"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F50%2Fpng&w=128&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fplacehold.co%2F50%2Fpng&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fplacehold.co%2F50%2Fpng&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
    <div
      class="ml-4 flex-1"
    >
      <p
        class="text-base font-semibold text-gray-800"
      >
        Company A
      </p>
      <div
        class="text-sm text-gray-500 line-clamp-2"
      >
        <p
          class="text-sm text-gray-500 "
        />
      </div>
    </div>
    <svg
      class="w-6 h-6 text-gray-400 ml-2 cursor-pointer"
      fill="none"
      stroke="currentColor"
      stroke-width="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 5l7 7-7 7"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </li>
</div>
`;
